tests: added test case for large transactions to transaction_tests.c

raffecat opened the pull request.

raffecat committed:
tests: added test case for large transactions to transaction_tests.c

edtubbs commented:
Thanks! Near-term fix is in #262. Test passes but triggers a format-string warning:
```
In file included from test/transaction_tests.c:13:
test/transaction_tests.c: In function ‘test_transaction’:
./include/test/utest.h:316:24: warning: too many arguments for format [-Wformat-extra-args]
  316 |                 printf("\tReceive:\tNULL\n", r_);                        \
      |                        ^~~~~~~~~~~~~~~~~~~~
test/transaction_tests.c:539:5: note: in expansion of macro ‘u_assert_not_null’
  539 |     u_assert_not_null(raw_hexadecimal_transaction);
      |     ^~~~~~~~~~~~~~~~~
```

raffecat commented:
Thanks Ed.

It's true, but the warnings come from the existing `u_assert_not_null` test macro, which I guess isn't being used anywhere else?

raffecat committed:
fix: removed extraneous argument from printf in `u_assert_not_null`

Looks like a copy-paste error, since `u_assert_is_null` has the same
argument and actually prints it.

edtubbs reviewed (approved):
Awesome, thanks!

michilumin commented:
CI failures of this current test are expected behavior with the current build.
